infrastructure/src/main/java/org/corfudb/infrastructure/BatchWriter.java

```diff
@@ -260,15 +261,15 @@ private void batchWriteProcessor() {
                                 res.add(currOp);
                                 break;
                             case TAILS_QUERY:
-                                TailsResponse tails = streamLog.getTails();
+                                TailsResponse tails = streamLog.getTails(currOp.getEpoch());
```

Comment:
I don't understand this. Why pass the message epoch all the way down to StreamLogs and then return it in the response? It seems like only BatchWriterOperation needs to know this, no?

Reply:
I agree. Since the TailsResponse is immutable, I did that to keep the construction of the TailsResponse in a single place. I'll change it.